NumberBaseball/Source.c

```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include <stdlib.h> 
+#include<time.h> 
+
+#define LINESIZE 1024
+#define NUMBER_OF_GUESSES 3
+#define STRIKES_TO_END 3
+
+int numStrikes;
+int numBalls;
+int numOuts;
+int resultStrikes;
+int resultBalls;
+int resultOuts;
+int n1;
+int n2;
+int n3;
+int random1;
+int random2;
+int random3;
+
+void Init() {
+	numStrikes = 0;
+	numBalls = 0;
+	numOuts = 0;
+	resultStrikes = 0;
+	resultBalls = 0;
+	resultOuts = 0;
+	printf("Init\n");
+}
+
+void Update() {
+	char line[LINESIZE];
+
+	resultStrikes = 0;
+	resultBalls = 0;
+	resultOuts = 0;
+
+	while (1) {
+		printf("Please enter your %d numbers:\n", NUMBER_OF_GUESSES);
+		if (!fgets(line, LINESIZE, stdin)) /* if we fail to read a line */
+		{
+			clearerr(stdin);
+			break;
```

Comment:
If a user types invalid input like characters..this code will let Render() print unnecessary information.

Reply:
Render should be checked by Update so it shouldn't be able to print unless there is valid information.

Reply:
Ohhh I see. I think that the only input that would cause this line to fail is EOF, Ctrl+Z (on windows). So since I changed it to an array that I make size equal to the number of guesses at the beginning, it just makes the user guess 0 0 0, which should be impossible. So, not gibberish anymore but not right. I'll modify it so they are prompted again.